lib/quill_delta.dart

```diff
@@ -463,6 +486,59 @@ class Delta {
     return result;
   }
 
+  /// invert this delta base on baseDelta
+  Delta invert(Delta base) {
+    Delta inverted = new Delta();
+
+    if (base.isEmpty) return inverted;
+    int baseIndex = 0;
+
+    for (Operation op in _operations) {
+      if (op.isInsert) {
+        inverted.delete(op.length);
+      } else if (op.isRetain && op.attributes == null) {
+        inverted.retain(op.length, null);
+        baseIndex += op.length;
+      } else if (op.isDelete || (op.isRetain && op.attributes != null)) {
+        var length = op.length;
+        Delta sliceDelta = base.slice(baseIndex, baseIndex + length);
+        sliceDelta.toList().forEach((baseOp) {
+          if (op.isDelete) {
+            inverted.push(baseOp);
+          } else if (op.isRetain && op.attributes != null) {
+            var invertAttr = invertAttributes(op.attributes, baseOp.attributes);
+            assert(invertAttr.isNotEmpty);
+            inverted.retain(baseOp.length, invertAttr);
+          }
+        });
+        baseIndex += length;
+      }
```

Comment:
I know this is not part of original implementation but any `else if` not followed by `else` make me worried. Could you add `else` branches here that throw an error like this:

```dart
} else {
  throw StateError("Unreachable");
}
```